### ISSUE TITLE
[Accessibility]A11y_ Java Tooling for IntelliJ/Eclipse – New Feature_Storage Container_Keyboard: Tooltips for the Non-text controls are not visible on keyboard focus

### ISSUE THREAD

**Issue by Tanushagottipati** [microsoft/azure-tools-for-java#3557]
[Check out Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Faccessibilityinsights.io%2F&data=02%7C01%7Cv-tagott%40microsoft.com%7Cb67b2c4b646d4f9561a208d6f4b5c39b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636965458847261333&sdata=PJMiKjb64uwhBN98LZpctnBiAVeOvw%2BLBrnxBtrjl2M%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier

GitHubTags: #A11yTCS; #A11yMAS; #DesktopApp; #Win10; #A11y_JavaToolingforIntelliJ/Eclipse–NewFeature; #MAS2.1.1; #Keyboard; #C+E-H; #JavatoolingforIntelliJ/Eclipse

**Environment Details:**
Eclipse IDE for Enterprise Java Developers.
Version: 2019-06 (4.12.0)
Build id: 20190614-1200
Solution Installed: The Azure Toolkit for Eclipse 3.17.0

**OS Details:** 
OS: Windows (Version 1903, Build: 18362.175)
MAS Violated: MAS2.1.1

**Note: Similar issue is found in** 
**1.	Spark Submission and** 
**2.	Spark Submission Tab**
**3.  Similar issue is observed in "Link New HDInsight Cluster" page for the controls 'Cluster Name', 'User Name', 'Password', 'Storage Account', 'Storage Key'**

**Repro Steps:** 
1) Open Java EE - Eclipse application
2) Install 'Azure Toolkit for Eclipse' from Marketplace
3) Open Eclipse and install the Scala plug-in 
4) Tab till tools-->Azure-->Azure Sign in and sign into Azure account and select any subscription
5) Tab till Windows menu option-->Show View-->Other-->Expand Azure-->Azure Explorer pane press OK
6) In Azure Explorer section, tab till HDInsight and expand it. Expand the cluster name and select Storage Container
7) In Storage Container View section, verify if Tooltips are visible for all the Non text controls on keyboard focus

**Actual Result:** 
While navigating with the help of keyboard tooltips present for the non-text controls are not visible when it receives keyboard focus

**Expected Result:** 
While navigating with the help of keyboard and on receiving keyboard focus on a non-text control or icon tooltip must be visible to the users

**User Impact:** 
Keyboard users will not be able to know the meaning or name of the non-text controls if tooltips are not visible

**MAS Reference:** 
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={8492c4eb-c179-40ae-8777-cd044ed725a2}

![Bug_3_Tooltips are not visible on keyboard focus](https://user-images.githubusercontent.com/41109135/65320877-797dc480-dbc0-11e9-98ea-c6dee0a57794.png)
[Bug_3_Tooltips are not visible on keyboard focus.pptx](https://github.com/microsoft/azure-tools-for-java/files/3635476/Bug_3_Tooltips.are.not.visible.on.keyboard.focus.pptx)
[Bug3_Tooltips are not provided for the keyboard users about the edit fields.pptx](https://github.com/microsoft/azure-tools-for-java/files/3656990/Bug3_Tooltips.are.not.provided.for.the.keyboard.users.about.the.edit.fields.pptx)

**Comment by t-rufang** [comment]
Note: Tooltip is not available when focus is at the component. The only way to see the tooltip is to put the mouse at the component
![image](https://user-images.githubusercontent.com/32627233/65827778-6eb0e700-e2c6-11e9-874a-50a0d54db1c5.png)
![Bug_3_Tooltips are not visible on keyboard focus](https://user-images.githubusercontent.com/41109135/65320877-797dc480-dbc0-11e9-98ea-c6dee0a57794.png)

**Comment by t-rufang** [comment]
All the UI components mentioned above are from Eclipse SWT UI components. This issue is a common issue if user use SWT to draw their UI. Please find another same issue in a dialog in Eclipse.
![issue_3557](https://user-images.githubusercontent.com/32627233/68654173-5809db00-0568-11ea-8638-03ebd593eadc.gif)
As you can see, when we focus at the down arrow button in the upper right position, the tool tip is not showing.
![image](https://user-images.githubusercontent.com/32627233/68654210-6657f700-0568-11ea-9292-0f6b30e323c8.png)
Therefore, this bug is owned by Eclipse and it's out of the scope of our toolkit.

**Comment by Flanker32** [comment]
@Tanushagottipati We have removed storage explorer in IDE toolkit, could you please help us verify and close the issue? Here is the [build](https://dev.azure.com/mseng/VSJava/_build/results?buildId=11079936)

**Comment by Tanushagottipati** [comment]
Regressed in the below environment and the  issue is fixed for 

- Spark Submission Tab - Here text is provided beside the non text controls
![Screenshot (710)](https://user-images.githubusercontent.com/41109135/70887759-b4976480-2004-11ea-9f9a-b5965324df5d.png)

- Spark Submission- here the 'Refresh' control has been removed
![Screenshot (711)](https://user-images.githubusercontent.com/41109135/70887803-cc6ee880-2004-11ea-990e-c28d75486816.png)

- Storage explorer- this page has been deprecated
![Screenshot (709)](https://user-images.githubusercontent.com/41109135/70887841-e4466c80-2004-11ea-9b25-752cdadc4413.png)

and this issue is not fixed for 'Link New HDInsight Cluster' where tooltips provided for the text fields are not appearing on keyboard focus. hence this issue is not fixed. hence reactivating this bug
[Tooltip issue is not fixed for link a cluster page.pptx](https://github.com/microsoft/azure-tools-for-java/files/3966564/Tooltip.issue.is.not.fixed.for.link.a.cluster.page.pptx)

Environment:
Eclipse IDE for Enterprise Java Developers.
Version: 2019-09 R (4.13.0)
Build id: 20190917-1200
Solution Installed: The Azure Toolkit for Eclipse 3.20.0

**Comment by t-rufang** [comment]
@Tanushagottipati Tooltip not visible issue should be fixed by Eclipse team, please find a similar situation in Eclipse Preferences/Gradle page.
Repro steps:
1. Open the Preferences dialog from menu bar Window -> Preferences.
2. Select Gradle tab from left side bar.
3. When the mouse hover the checkbox, there are tooltips shown on the checkbox. But whenwe use keyboard to tab to these checkboxs, there are no tooltips shown.
![issue_3557](https://user-images.githubusercontent.com/32627233/70889367-6b074380-201d-11ea-9057-b490a2173499.gif)

**Comment by t-rufang** [comment]
- Importance of experience: Important. Keyboard user can't see the tooltip when they tab to the text field, therefore they might find it hard to understand what the meaning of the text field is.
- Quality of experience: Manageable. User can still fill in the text fields without hint.
- Alternate solutions: No alternate technologies or options can be leveraged.
- Evidence: All evidence is provided in this issue.
- [AAT Link](https://cela-ra-aat-prod.azurewebsites.net/#/workflow/56452990-0d38-45bc-b475-a62832ce0076)

**Comment by Tanushagottipati** [comment]
@Alexandria-R
Could you please confirm if we can close this bug based on the above comments. This issue is resolved as external to Eclipse

**Comment by Alexandria-R** [comment]
This seems like a valid issue on Eclipse's side. Go ahead, @Tanushagottipati

**Comment by Tanushagottipati** [comment]
Closing this bug based on the above comments

**Comment by vscjavaci** [comment]
Track internal issue by AB#1749667
